pkgs/development/compilers/gcc/common/libgcc.nix

```diff
@@ -0,0 +1,88 @@
+{ lib
+, stdenv
+, langC
+, langCC
+, langJit
+}:
+
+let
+  enableLibGccOutput = (with stdenv; targetPlatform == hostPlatform) && !langJit;
+in
+(pkg: pkg.overrideAttrs (previousAttrs: lib.optionalAttrs ((!langC) || langJit || enableLibGccOutput) {
+  outputs = previousAttrs.outputs ++ lib.optionals enableLibGccOutput [ "libgcc" ];
+  # This is a separate phase because gcc assembles its phase scripts
+  # in bash instead of nix (we should fix that).
+  preFixupPhases = (previousAttrs.preFixupPhases or []) ++ [ "postPostInstallPhase" ];
+  postPostInstallPhase =
+    # delete extra/unused builds of libgcc_s in non-langC builds
+    # (i.e. libgccjit, gnat, etc) to avoid potential confusion
+    lib.optionalString (!langC) ''
+      rm -f $out/lib/libgcc_s.so*
+    ''
+    # move `libgcc_s.so` into its own output, `$libgcc`
+    + lib.optionalString enableLibGccOutput (''
+      # move libgcc from lib to its own output (libgcc)
+      mkdir -p $libgcc/lib
+      mv    $lib/lib/libgcc_s.so      $libgcc/lib/
+      mv    $lib/lib/libgcc_s.so.1    $libgcc/lib/
+      ln -s $libgcc/lib/libgcc_s.so   $lib/lib/
+      ln -s $libgcc/lib/libgcc_s.so.1 $lib/lib/
```

Comment:
In case of `firefox / lld` you noticed that you need to pass some form of a path to `libgcc_s.so` and add `-L${gccForLibs.libgcc}/lib` to the `cc-wrapper`.

Here you effectively make `$libgcc/lib` and `$lib/lib/` interchangeable for that purpose.

From this my understanding is that `$libgcc/lib` is the preferred path we want to use. I suspect it will trigger addition of `RUNPATH` entries to `$libgcc/lib` to every binary. I think this all is fine.

In light of the above do we need these `ln -s` symlinks shorter and longer term? If we remove them today (or later) will more things break or we could just do it?

Reply:
We should drop them now if possible.  I'm verifying that right now.

Reply:
Follow-up?

Reply:
See https://github.com/NixOS/nixpkgs/pull/209870#issuecomment-1493235853

Resolving for now (deferred as https://github.com/NixOS/nixpkgs/issues/221525)